crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -475,6 +546,18 @@ impl Mesh {
         Ok(())
     }
 
+    /// Consumes the mesh and returns a mesh with tangents generated using the `mikktspace` algorithm.
+    ///
+    /// The resulting mesh will have the [`Mesh::ATTRIBUTE_TANGENT`] attribute if successful.
+    ///
+    /// (Alternatively, you can use [`Mesh::generate_tangents`] to mutate an existing mesh in-place)
+    ///
+    /// Requires a [`PrimitiveTopology::TriangleList`] topology and the [`Mesh::ATTRIBUTE_POSITION`], [`Mesh::ATTRIBUTE_NORMAL`] and [`Mesh::ATTRIBUTE_UV_0`] attributes set.
+    pub fn with_generated_tangents(mut self) -> Result<Mesh, GenerateTangentsError> {
```

Comment:
#[must_use] ?

Reply:
I originally had it here too, but clippy complained about the redundancy with `Result<>`, which is already `#[must_use]`.

Reply:
Oh, fair! Interesting interaction